toffy/qc_comp_test.py

```diff
@@ -178,129 +181,170 @@ def test_compute_99_9_intensity():
         )
-        misc_utils.verify_same_elements(
-            provided_chans=chans,
-            nzm_chans=intensity_99_9_batch.drop(columns='fov').columns.values
+        percentile_99_9_path = os.path.join(
+            bin_file_path, '%s_percentile_99_9_stats.csv' % fovs[0]
         )
 
+        # assert the QC files for fov-1-scan-1 were created
+        assert os.path.exists(nonzero_mean_path)
+        assert os.path.exists(total_intensity_path)
+        assert os.path.exists(percentile_99_9_path)
+
+        # read in the QC data
+        nm = pd.read_csv(nonzero_mean_path)
+        ti = pd.read_csv(total_intensity_path)
+        p99_9 = pd.read_csv(percentile_99_9_path)
+
+        # assert the column names are correct
+        assert list(nm.columns.values) == ['fov', 'channel', 'Non-zero mean intensity']
+        assert list(ti.columns.values) == ['fov', 'channel', 'Total intensity']
+        assert list(p99_9.columns.values) == ['fov', 'channel', '99.9 intensity value']
+
+        # assert the correct FOV was written
+        assert list(nm['fov']) == [fovs[0]]
+        assert list(nm['fov']) == list(ti['fov']) == list(p99_9['fov'])
+
+        # assert the correct channels were written
+        assert list(nm['channel']) == ['SMA']
+        assert list(nm['channel']) == list(ti['channel']) == list(p99_9['channel'])
 
-@pytest.mark.parametrize("test_fovs,test_chans,test_gaussian_blur", FOVS_CHANS_TEST_QC)
-def test_compute_qc_metrics(test_fovs, test_chans, test_gaussian_blur):
+
+@parametrize('fovs', [['fov-1-scan-1'], ['fov-1-scan-1', 'fov-2-scan-1']])
+def test_combine_qc_metrics(fovs):
     with tempfile.TemporaryDirectory() as temp_dir:
-        # define 3 fovs and 3 channels
-        fovs, chans = test_utils.gen_fov_chan_names(3, 3)
-
-        # make the sample data
-        tiff_dir = os.path.join(temp_dir, "single_channel_inputs")
-        img_sub_folder = "TIFs"
-
-        os.mkdir(tiff_dir)
-        test_utils.create_paired_xarray_fovs(
-            base_dir=tiff_dir,
-            fov_names=fovs,
-            channel_names=chans,
-            img_shape=(40, 40),
-            sub_dir=img_sub_folder,
-            fills=True,
-            dtype="int16"
-        )
+        # define a dummy list of channels
+        chans = ['SMA', 'Vimentin', 'Au']
+
+        # define a sample bin_file_path
+        bin_file_path = os.path.join(temp_dir, 'sample_qc_dir')
+        os.mkdir(bin_file_path)
+
+        # put some random stuff in bin_file_path, test that this does not affect aggregation
+        pd.DataFrame().to_csv(os.path.join(bin_file_path, 'random.csv'))
+        Path(os.path.join(bin_file_path, 'random.txt')).touch()
+
+        # add existing combined .csv files, this also should not affect aggregation
+        pd.DataFrame().to_csv(os.path.join(bin_file_path, 'combined_nonzero_mean_stats.csv'))
+        pd.DataFrame().to_csv(os.path.join(bin_file_path, 'combined_total_intensity_stats.csv'))
```

Comment:
Why do we need this at the beginning?

Reply:
We write these blank `combined.csv` files to ensure that even if `mibi_bin_path` contains them, `combine_qc_metrics` won't try to grab these files when aggregating. Specifically, it tests this line: `metric_files = [mf for mf in metric_files if 'combined' not in mf]`.